modern/src/reports/ChartReportPage.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import { Box, Paper } from '@material-ui/core';
+import ReportFilter from './ReportFilter';
+import ReportLayoutPage from './ReportLayoutPage';
+import { useAttributePreference } from '../common/preferences';
+import { formatDate } from '../common/formatter';
+import { getConverter } from '../common/converter';
+import {LineChart, Line, XAxis, YAxis, CartesianGrid, Tooltip, Legend, ResponsiveContainer } from 'recharts';
+
+const ReportFilterForm = ({ setItems }) => {
+
+  const speedUnit = useAttributePreference('speedUnit');
+
+  const handleSubmit = async (deviceId, from, to, mail, headers) => {
+    const query = new URLSearchParams({ deviceId, from, to, mail });
+    const response = await fetch(`/api/reports/route?${query.toString()}`, { headers });
+    if (response.ok) {
+
+      const data = await response.json();
+
+      let formattedData = data.map(obj => {
+        return Object.assign(
```

Comment:
I don't like reusing the same model. Let's create a separate new object with only the data that we need.

Reply:
done